ecommerce/api.py

```diff
@@ -379,3 +379,40 @@ def is_coupon_redeemable(coupon, user):
         return any((mmtrack.has_verified_cert(run.edx_course_key) for run in course.courserun_set.all()))
 
     return True
+
+
+def pick_coupons(user):
+    """
+    Choose the coupon which would be used in redemptions by the user.
+
+    The heuristic is currently:
+     - choose attached coupons over automatic coupons
+     - choose the coupon which has been most recently attached, or most recently modified
+
+    Args:
+        user (django.contrib.auth.models.User): A user
+
+    Returns:
+        Coupon: The coupon which will be used by the user when redeeming runs in a program
+    """
+    sorted_attached_coupons = list(
+        Coupon.objects.filter(usercoupon__user=user).order_by('-usercoupon__updated_on')
+    )
+    sorted_automatic_coupons = list(
+        Coupon.objects.filter(Coupon.is_automatic_qset()).order_by('-updated_on')
```

Comment:
I think that `Coupon.is_automatic_qset()` needs to be at least renamed given that it does not return a queryset.

BUT, given that it is used always in a `.filter`, I think it should be refactored to do the actual filter and return a queryset, so here and anywhere else can be just be refined.
This is up to you, but if you choose not to refactor, please rename it.